tutorials/io/plot_20_reading_eeg_data.py

```diff
@@ -161,45 +164,74 @@
+EEG data from the Persyst system can be read with
+:func:`mne.io.read_raw_persyst`.
+
+Note that not all subject metadata may be properly imported because Persyst
+sometimes changes its specification from version to version. Please let us know
+if you encounter a problem.
 
-Note that not all the subject metadata may be properly read in
-due to the fact that Persyst changes its specification
-sometimes from version to version. Please submit an issue, or
-pull request if you encounter a problem.
 
-Nihon Kohden EEG data (.EEG, .21E, .PNT, .LOG)
+Nihon Kohden EEG data (.eeg, .21e, .pnt, .log)
 ==============================================
 
 EEG data from the Nihon Kohden (NK) system can be read using the
 :func:`mne.io.read_raw_nihon` function.
 
 Files with the following extensions will be read:
 
-- The ``.EEG`` file contains the actual raw EEG data.
-- The ``.PNT`` file contains the metadata related to the recording, such
-  as the measurement date.
-- The ``.LOG`` file contains annotations for the recording.
-- The ``.21E`` file contains the channel and electrode
-  recording system information.
+- The ``.eeg`` file contains the actual raw EEG data.
+- The ``.pnt`` file contains metadata related to the recording such as the
+  measurement date.
+- The ``.log`` file contains annotations for the recording.
+- The ``.21e`` file contains channel and electrode information.
 
-Reading ``.11D``, ``.CMT``, ``.CN2``, and ``.EDF`` files is currently not
+Reading ``.11d``, ``.cmt``, ``.cn2``, and ``.edf`` files is currently not
 supported.
 
-Note that not all the subject metadata may be properly read in
-due to the fact that NK changes the specification
-sometimes from version to version. Please submit an issue, or
-pull request if you encounter a problem.
+Note that not all subject metadata may be properly read because NK changes the
+specification sometimes from version to version. Please let us know if you
+encounter a problem.
+
+
+XDF data (.xdf, .xdfz)
+======================
+
+MNE-Python does not support loading
+`XDF <https://github.com/sccn/xdf/wiki/Specifications>`_ files out of the box,
+because the inherent flexibility of the XDF format makes it difficult to
+provide a one-size-fits-all function. For example, XDF supports signals from
+various modalities recorded with different sampling rates. However, it is
+relatively straightforward to import only a specific stream (such as EEG
+signals) using the `pyxdf <https://github.com/xdf-modules/pyxdf>`_ package. The
+following code snippet demonstrates the basic mechanisms how this could work:
+
+.. code-block:: python
+
+   import pyxdf
+
+   streams, header = pyxdf.load_xdf(filename)  # replace with actual file name
+   data = streams[0]["time_series"].T
+   n_chans = int(streams[0]["info"]["channel_count"][0])
+   sfreq = float(streams[0]["info"]["nominal_srate"][0])
+   info = mne.create_info(n_chans, sfreq, "eeg")
+   raw = mne.io.RawArray(data, info)
```

Comment:
This will not get executed (only formatted/highlighted) and so therefore it's continued correct functioning won't get tested upon doc builds.  (I realize it's not meant to be executed, since you don't provide a filename, but I strongly prefer to avoid code samples in our docs that aren't actually run/tested).  I prefer either:

1. update this code so it works and end the triple-quoted docstring just before this code block so that this gets executed, or
2. follow my previous suggestion (https://github.com/mne-tools/mne-python/issues/5180#issuecomment-810371278) and add this content as a very short I/O example and crossreference it from within this tutorial.

perhaps it goes without saying that I prefer option 2.

Reply:
Where do you want me to add this as a very short I/O example? Can you point me to the LSL example? The [NEO example](https://mne.tools/dev/auto_examples/io/plot_read_neo_format.html) (I think that's what you meant) creates toy data, so it looks like we will have to add a short XDF sample file somewhere (where?).

Reply:
> Where do you want me to add this as a very short I/O example?

in `doc/examples/io/plot_xx_some_new_filename.py`

> Can you point me to the LSL example?

That is what I'm asking you to create.  I was being a bit careless between "LSL" and "XDF" because XDF is the default format for the LSL recorder.

> it looks like we will have to add a short XDF sample file somewhere (where?)

MNE-testing-data?  Or put a small file on OSF?  Or use one that is [already on OSF](https://osf.io/search/?q=xdf&filter=file&page=1)?  @larsoner maybe has stronger feelings than I do about where that should live.

Reply:
If it's not used in testing, don't put it in `mne-testing-data`. If it's small (< 50 MB?) it can live in https://github.com/mne-tools/mne-misc-data . If it's big it should get its own datasets downloader probably.

Reply:
OK. Does anyone have a small example file? I think it would be good to have something that contains a few seconds of EEG and maybe even some markers. There's a sample file [here](https://github.com/xdf-modules/example-files/blob/master/data_with_clock_resets.xdf), but the EEG data it contains is incorrectly scaled:
<img width="1113" alt="Screen Shot 2021-04-13 at 18 18 47" src="https://user-images.githubusercontent.com/4377312/114586233-bb7bc900-9c84-11eb-82b0-196902a7b96c.png">

Reply:
Not sure if it helps, but I have a normal-sized (I think ~30min recording of a small EEG stream) example file from my "eeg_matchingpennies" dataset: https://osf.io/cj2dr/

see e.g., https://osf.io/agj2q/

that could be truncated, or load as a whole from OSF :man_shrugging: 

you could also get it with datalad, I have a git-annex mirror here: https://github.com/sappelhoff/eeg_matchingpennies

Reply:
I think this file is too large (>500MB) so we'd need to truncate and then store it somewhere else. However, I don't know how to truncate an XDF file.

Reply:
@tstenner @cboulay do you happen to have a short demo XDF file, ideally with a few seconds of EEG, which we could use here? Alternatively, do you know how we could create such a file (e.g. crop existing XDF, create XDF with toy data)?

Reply:
In the coming days I can record some using an actiChamp LSL connector / streamer paired with the standard LSL stream recorder (which outputs XDF)